### DISCUSSION TITLE
Spark Essentials Difficulty

### DISCUSSION THREAD

**Discussion by andreilgeorgescu** [riverwalk-software/astro-test#81] [Q&A]
@daniel-ciocirlan I see that its difficulty is marked as `beginner` despite the fact that the rest of the essentials courses are `intermediate`. Is this intentional?

**Comment by daniel-ciocirlan** [comment]
I think it's best to keep the essentials courses to "beginner" instead.

The question is legit but it has a more complex answer, because the difficulty of the courses varies _at the stage where one would take them_. 

For example, Scala Essentials is tougher than Spark Essentials, even as Spark requires Scala. That's because at their respective learning stage, learning a different mindset (Scala) is harder than _having known_ Scala and taking Spark (which is a breeze once you know Scala basics).

That's the reasoning behind my labels, but they might confuse people seeing them, so easier to set all Essentials courses to "beginner" instead.

**Reply by andreilgeorgescu** [reply]
I see what you are saying. It is difficult because in the absolute, even a beginner Scala course is more for intermediate developers. 

However, as you said I think it makes more sense to label them relative to the topic. This is similar to how you'd see a linear algebra textbook be called "introductory" even though it's clearly not "beginner math".

I think it comes down to what these labels are referring to:

- the skill level of the user
  - not very reliable or useful
- how much prerequisite knowledge is required to know in that specific topic
  - useful but can be confusing since a "light speed" course would be labeled the same as a "zero to hero" kind of course
- how deep into the topic it goes
  - useful but people will think zero to hero courses aren't for them even when they are because they'll just say "advanced"

I think the best of both worlds would be having the difficulty represent a range from what you need to know to how deep the course goes in that topic. For example, a light speed course could just be "introductory", while a zero to hero would be "introductory - mastery".

And to top it off having pre-requisites that can link to what courses are required would allow people to better understand what they need to know prior to taking the course as well as advertise your other courses.

For example, the ZIO course could have a section that lists out requirements like "Scala essentials, advanced scala, etc" and link to them. That could also allow for more bundles such as including all the requirements in a bundle.

This is probably something for after the MVP but I think it would be a big improvement in usability and sales and also clean up the FAQs that talk about difficulty/prerequisites.

What do you think?
